realm/realm-library/src/main/java/io/realm/exceptions/RealmFileException.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.exceptions;
+
+import io.realm.internal.Keep;
+import io.realm.internal.SharedRealm;
+
+/**
+ * Class for reporting problems when the accessing Realm related files.
+ */
+@Keep
+public class RealmFileException extends RuntimeException {
+    /**
+     * The specific kind of this {@link RealmFileException}.
+     */
+    public enum Kind {
+        /**
+         * Thrown for any I/O related exception scenarios when a realm is opened.
+         */
+        ACCESS_ERROR,
+        /**
+         * Thrown if the user does not have permission to open or create the specified file in the specified access
+         * mode when the realm is opened.
+         */
+        PERMISSION_DENIED,
+        /**
+         * Not used now.
+         * Thrown if create_Always was specified and the file did already exist when the realm is opened.
```

Comment:
This is extremely internal, perhaps something like "Thrown if the file already existed, but wasn't supposed to" or something like that